build/package/scripts/cache-build.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+set -eu
+# Copy build to cache
+
+
+# the copy commands are based on GNU cp tools
+# On a mac `brew install coreutils` gives `g` prefixed cmd line tools such as gcp
+# to use these define env variable GNU_CP=gcp before invoking this script.
+CP="${GNU_CP:-cp}"
+
+OUTPUT_DIR="docker"
+WORKING_DIR="."
+CACHE_BUILD_KEY=
+CACHE_LOCATION_USED_PATH=
+DEBUG="${DEBUG:-false}"
+
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+
+    --working-dir) WORKING_DIR="$2"; shift;;
+    --working-dir=*) WORKING_DIR="${1#*=}";;
+
+    --output-dir) OUTPUT_DIR="$2"; shift;;
+    --output-dir=*) OUTPUT_DIR="${1#*=}";;
+
+    --cache-build-key) CACHE_BUILD_KEY="$2"; shift;;
+    --cache-build-key=*) CACHE_BUILD_KEY="${1#*=}";;
+
+    --cache-location-used-path) CACHE_LOCATION_USED_PATH="$2"; shift;;
+    --cache-location-used-path=*) CACHE_LOCATION_USED_PATH="${1#*=}";;
+
+    --debug) DEBUG="$2"; shift;;
+    --debug=*) DEBUG="${1#*=}";;
+
+  *) echo "Unknown parameter passed: $1"; exit 1;;
+esac; shift; done
+
+if [ -z "${CACHE_BUILD_KEY}" ]; then
+  echo "Param --cache-build-key is required."; exit 1;
+elif [ -z "${CACHE_LOCATION_USED_PATH}" ]; then
+  echo "Param --cache-location-used-path is required."; exit 1;
+fi
+
+if [ "${DEBUG}" == "true" ]; then
+  set -x
+fi
+
+ROOT_DIR=$(pwd)
+
+if [ "${WORKING_DIR}" == "." ]; then
+  git_sha_working_dir=$(git rev-parse "HEAD:")
+else
+  git_sha_working_dir=$(git rev-parse "HEAD:$WORKING_DIR")
+fi
+cache_location_dir="$ROOT_DIR/.ods-cache/build-task/$CACHE_BUILD_KEY/$git_sha_working_dir"
+
+if [ "${WORKING_DIR}" != "." ]; then
+  cd "${WORKING_DIR}"
+fi
+
+rm -rvf "$cache_location_dir"  # should be empty as otherwise cache should be used.
+mkdir -p "$cache_location_dir"
+
+# Copying reports
```

Comment:
Why is this talking about `reports` rather than `artifacts`? I am a bit confused about the language here.

Reply:
In past experiences the term artifacts included the build binaries, jars and similar files as well as related metafiles enabling further deployment. So for me the term artifacts appeared to generic when writing the code. But I can also live with changing the language and perhaps referencing the artifacts.adoc  in the comments.

Reply:
OK. I would like to use `artifacts` in this PR to avoid any confusion.

However, I think I agree with you that `reports` would have been the better name for the general concept. How about opening a new issue, where we can decide if changing to `reports` everywhere is the right way to go, and then change the naming in one go everywhere?

Reply:
I changed so that now the term artifacts is used instead of reports.